src/Storefront/Resources/app/storefront/test/plugin/forms/form-ajax-submit.plugin.test.js

```diff
@@ -145,12 +155,91 @@ describe('FormAjaxSubmitPlugin tests', () => {
+        // Add required attribute to input & Remove value from input
+        document.body.innerHTML = `
+            <div class="replace-me"></div>
+
+            <form method="post" action="/account/newsletter/subscribe">
+                <input type="email" name="email" required>
+                <button>Subscribe to newsletter</button>
+            </form>
+        `;
+
+        formElement = document.querySelector('form');
+
+        formAjaxSubmit = new FormAjaxSubmitPlugin(formElement, {
+            replaceSelectors: ['.replace-me'],
+            submitOnChange: true,
+            noValidate: true,
+        });
+
+        global.fetch = jest.fn(() =>
+            Promise.resolve({
+                text: () => Promise.resolve('<div class="replace-me"><div class="alert">Success</div></div>'),
+            })
+        );
+
+        formAjaxSubmit.$emitter.publish = jest.fn();
+
+        window.PluginManager.initializePlugins = jest.fn();
+
+        // execute the tests on the invalid form
+        const cb = jest.fn();
+        const inputElement = formElement.querySelector('input');
+
+        formAjaxSubmit.addCallback(cb);
+        // native browser change event not cancelable
+        inputElement.dispatchEvent(new Event('change', { bubbles: true, cancelable: false }));
+        await new Promise(process.nextTick);
+
+        expect(cb).toHaveBeenCalledTimes(1);
+    });
+
+    test('form uses correct formAction & formMethod from submitter', async () => {
+        // Add formAction & formMethod to submitter
+        document.body.innerHTML = `
+            <div class="replace-me"></div>
+
+            <form method="post" action="/account/newsletter/subscribe">
+                <input type="email" name="email" value="test@example.com">
+                <button type="submit" formaction="/account/newsletter/unsubscribe" formmethod="get">Subscribe to newsletter</button>
+            </form>
+        `;
+
+        formElement = document.querySelector('form');
+
+        formAjaxSubmit = new FormAjaxSubmitPlugin(formElement, {
+            replaceSelectors: ['.replace-me'],
+            submitOnChange: true,
+        });
+
+        global.fetch = jest.fn((url, options) => {
+            expect(url).toBe('/account/newsletter/unsubscribe');
+            expect(options).toStrictEqual({ headers: { 'X-Requested-With': 'XMLHttpRequest' }});
+
+            return Promise.resolve({
+                text: () => Promise.resolve('<div class="replace-me"><div class="alert">Success</div></div>'),
+            });
+        });
+
+        formAjaxSubmit.$emitter.publish = jest.fn();
+
+        window.PluginManager.initializePlugins = jest.fn();
+
+        const event = new Event('submit');
+        event.submitter = formElement.querySelector('button');
+
+        formElement.dispatchEvent(event);
+        await new Promise(process.nextTick);
+
+        expect(global.fetch).toHaveBeenCalledTimes(1);
```

Comment:
This test does not really assert that the correct formAction and formMethod are used.